core/src/main/scala/org/apache/spark/config/ConfigBuilder.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.config
```

Comment:
I'm a little confused. Wouldn't `internal.config` also show up in the docs? What makes something do or do not show up in docs? What's the new convention for adding new private Spark types to the code base?

Everything here is `private[spark]`, except for the package objects which Scala doesn't allow being `private` anything. So what would show up in the docs here and potentially confuse people?